components/forms/CreateTxForm/MsgForm/MsgCreateVestingAccountForm.tsx

```diff
@@ -0,0 +1,190 @@
+*/
+const getMinEndTime = (): string => {
+  const minTimestamp = Date.now() + 1000 * 60 * 60 * 24 * 30;
+  const minDate = new Date(minTimestamp);
+
+  const minMonth = minDate.getMonth() + 1; // It's 0-indexed
+  const minMonthStr = minMonth < 10 ? `0${minMonth}` : String(minMonth);
+
+  const minDay = minDate.getDate();
+  const minDayStr = minDay < 10 ? `0${minDay}` : String(minDay);
+
+  const minHours = minDate.getHours();
+  const minHoursStr = minHours < 10 ? `0${minHours}` : String(minHours);
+
+  const minMinutes = minDate.getMinutes();
+  const minMinutesStr = minMinutes < 10 ? `0${minMinutes}` : String(minMinutes);
+
+  return `${minDate.getFullYear()}-${minMonthStr}-${minDayStr}T${minHoursStr}:${minMinutesStr}`;
+};
+
+interface MsgCreateVestingAccountFormProps {
+  readonly fromAddress: string;
+  readonly setMsgGetter: (msgGetter: MsgGetter) => void;
+  readonly deleteMsg: () => void;
+}
+
+const MsgCreateVestingAccountForm = ({
+  fromAddress,
+  setMsgGetter,
+  deleteMsg,
+}: MsgCreateVestingAccountFormProps) => {
+  const { state } = useAppContext();
+  assert(state.chain.addressPrefix, "addressPrefix missing");
+
+  const [toAddress, setToAddress] = useState("");
+  const [amount, setAmount] = useState("0");
+  const minEndTime = getMinEndTime();
+  const [endTime, setEndTime] = useState(minEndTime);
+  const [delayed, setDelayed] = useState(true);
+
+  const [toAddressError, setToAddressError] = useState("");
+  const [amountError, setAmountError] = useState("");
+  const [endTimeError, setEndTimeError] = useState("");
+
+  useEffect(() => {
+    try {
+      assert(state.chain.denom, "denom missing");
+
+      setToAddressError("");
+      setAmountError("");
+      setEndTimeError("");
+
+      const isMsgValid = (msg: TxMsg): msg is TxMsgCreateVestingAccount => {
+        assert(state.chain.addressPrefix, "addressPrefix missing");
+
+        const addressErrorMsg = checkAddress(toAddress, state.chain.addressPrefix);
+        if (addressErrorMsg) {
+          setToAddressError(
+            `Invalid address for network ${state.chain.chainId}: ${addressErrorMsg}`,
+          );
+          return false;
+        }
+
+        if (!amount || Number(amount) <= 0) {
+          setAmountError("Amount must be greater than 0");
+          return false;
+        }
+
+        if (!endTime) {
+          setEndTimeError("End time is required");
+          return false;
+        }
+
+        return isTxMsgCreateVestingAccount(msg);
+      };
+
+      const amountInAtomics = amount
+        ? Decimal.fromUserInput(amount, Number(state.chain.displayDenomExponent)).atomics
+        : "0";
+
+      const msg: TxMsgCreateVestingAccount = {
+        typeUrl: "/cosmos.vesting.v1beta1.MsgCreateVestingAccount",
+        value: {
+          fromAddress,
+          toAddress,
+          amount: [{ amount: amountInAtomics, denom: state.chain.denom }],
+          endTime: timestampFromDatetimeLocal(endTime),
+          delayed,
+        },
+      };
```

Comment:
This `msg` looks alright here. You can add type-dafety if you store the `value: MsgCreateVestingAccount` in an extra line.

But how does the object look like when it was written to Fauna and read from Fauna as part of the signing process? I'd assume this breaks the Long.

```
      const {
        data: { transactionID },
      } = await axios.post("/api/transaction", {
        dataJSON: JSON.stringify(tx),
      });
  ```
  
  writed the whole tx including the messages to JSON. There it looks like this:
  
  ```
  > const Long = require("long")
undefined
> Long.fromNumber(123)
Long { low: 123, high: 0, unsigned: false }
> let a = { endTime: Long.fromNumber(123) }
undefined
> JSON.stringify(a)
'{"endTime":{"low":123,"high":0,"unsigned":false}}'
```

Then in `  const txInfo: DbTransaction = transactionJSON ? JSON.parse(transactionJSON) : null;` this becomes something like

> > JSON.parse('{"endTime":{"low":123,"high":0,"unsigned":false}}')
{ endTime: { low: 123, high: 0, unsigned: false } }.

I'm a bit surprised the proto encoding does not fail entirely.

How does `props.tx.msgs` look like in this code block?

```
      const { bodyBytes, signatures } = await signingClient.sign(
        signerAddress,
        props.tx.msgs,
        props.tx.fee,
        props.tx.memo,
        signerData,
      );
```

Reply:
`props.tx.msgs` looks like this:

![image](https://github.com/cosmos/cosmos-multisig-ui/assets/44572727/5fa5cc0c-a95b-42b7-8981-308f28790eab)

I made sure that only the methods go away in serialization, so when reading the tx from db I re-build the Long obj with `Long.fromValue` to be able to use `longObj.multiply(1000)`